### PR TITLE
Restrict access to Clock Stops and Open Pathways

### PR DESCRIPTION
**Pull request by iaindillingham**
See opensafely-core/ehrql#1646 and opensafely-core/ehrql#1651.

### CODE REVIEW

File: tests/fixtures/permissions/repository_permissions.yaml [hunk 1]
```diff
@@ -12,5 +12,9 @@ opensafely/dummy_isaric:
   allow: ['isaric']
 opensafely/dummy_open_prompt:
   allow: ['open_prompt']
+opensafely/dummy_wl_clockstops:
+  allow: ['wl_clockstops']
+opensafely/dummy_wl_openpathways:
+  allow: ['wl_openpathways']
 opensafely/dummy_all:
-  allow: ['icnarc', 'ons', 'isaric', 'ons_cis', 'ukrr', 'open_prompt']
+  allow: ['icnarc', 'ons', 'isaric', 'ons_cis', 'ukrr', 'open_prompt', 'wl_clockstops', 'wl_openpathways']
```
Comment (rebkwok) [inline tests/fixtures/permissions/repository_permissions.yaml:20]: Should we take ons/ons_cis out of the tests too?

File: opensafely/check.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ class RestrictedDataset:
         cohort_extractor_function_names=[
             "with_an_ons_cis_record",
         ],
-        ehrql_table_names=["ons_cis_raw", "ons_cis"],
+        ehrql_table_names=[],
```
Comment (rebkwok) [inline opensafely/check.py:44]: Should we remove the entire `RestrictedDataset`?  The function `with_an_ons_cis_record` no longer exists in cohort-extractor; if we're removing the ehql references, we probably should do the same for cohort-extractor

**Review by rebkwok — APPROVED** [review]

File: tests/fixtures/permissions/repository_permissions.yaml (hunk 1 above)
Comment (iaindillingham) [reply tests/fixtures/permissions/repository_permissions.yaml]: We should! Thank you for spotting this.

File: opensafely/check.py (hunk 2 above)
Comment (iaindillingham) [reply opensafely/check.py]: We should! I hadn't realized -- or more likely, I had forgotten -- that we removed ONS-CIS from cohort-extractor too.

**Pull request merged**